zivid_camera/src/zivid_camera.cpp

```diff
@@ -167,38 +190,31 @@ ZividCamera::ZividCamera(ros::NodeHandle& nh, ros::NodeHandle& priv)
   ROS_INFO_STREAM(camera_);
   if (!file_camera_mode)
   {
-    ROS_INFO_STREAM("Connecting to camera '" << camera_.serialNumber() << "'");
+    ROS_INFO_STREAM("Connecting to camera '" << camera_.info().serialNumber() << "'");
     camera_.connect();
   }
-  ROS_INFO_STREAM("Connected to camera '" << camera_.serialNumber() << "'");
+  ROS_INFO_STREAM("Connected to camera '" << camera_.info().serialNumber() << "'");
   setCameraStatus(CameraStatus::Connected);
 
   camera_connection_keepalive_timer_ =
       nh_.createTimer(ros::Duration(10), &ZividCamera::onCameraConnectionKeepAliveTimeout, this);
 
-  const auto defaultSettings = camera_.settings();
-  capture_general_config_dr_server_ =
-      std::make_unique<CaptureGeneralConfigDRServer>("capture/general", nh_, defaultSettings);
-
-  ROS_INFO("Setting up %d capture/frame_<n> dynamic_reconfigure servers", num_capture_frames);
-  for (int i = 0; i < num_capture_frames; i++)
-  {
-    capture_frame_config_dr_servers_.push_back(
-        std::make_unique<CaptureFrameConfigDRServer>("capture/frame_" + std::to_string(i), nh_, defaultSettings));
-  }
+  capture_settings_controller_ =
+      std::make_unique<Capture3DSettingsController>(nh_, camera_, "settings", max_capture_acquisitions);
```

Comment:
Much tidier factoring out the settings controller related code from the `zivid_camera` :+1: